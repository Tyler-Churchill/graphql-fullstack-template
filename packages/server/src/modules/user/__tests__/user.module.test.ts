import { gql } from 'apollo-server-core';
import TestClient from '../../../tests/TestClient';

it('Can register by email', async () => {
  const resp = await TestClient.mutate({
    mutation: gql`
      mutation {
        registerByEmail(email: "ttwwt@awadw.com", password: "testtest") {
          email
          firstName
          lastName
          isEmailVerified
          roles
        }
      }
    `
  });
  expect(resp).toMatchSnapshot();
});

it('Can login by email', async () => {
  const email = 'testemail@email.com';
  const password = 'testtest';
  await TestClient.mutate({
    mutation: gql`
      mutation {
        registerByEmail(email: "${email}", password: "${password}") {
          email
          firstName
          lastName
          isEmailVerified
          roles
        }
      }
    `
  });
  const resp = await TestClient.mutate({
    mutation: gql`
      mutation {
        loginByEmail(email: "${email}", password: "${password}") {
          email
        }
      }
    `
  });
  expect(resp).toMatchSnapshot();
});
