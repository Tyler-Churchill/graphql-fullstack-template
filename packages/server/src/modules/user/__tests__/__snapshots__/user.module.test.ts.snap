// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can register by email 1`] = `
Object {
  "data": Object {
    "registerByEmail": Object {
      "__typename": "User",
      "email": "ttwwt@awadw.com",
      "firstName": null,
      "isEmailVerified": false,
      "lastName": null,
      "roles": Array [
        "PUBLIC",
      ],
    },
  },
}
`;
